cpg_workflows/jobs/simple_vep_annotation.py

```diff
@@ -0,0 +1,206 @@
+
+        # declare a resource group for this annotated VCF output
+        vep_job.declare_resource_group(vcf={'vcf.bgz': '{root}.vcf.bgz', 'vcf.bgz.tbi': '{root}.vcf.bgz.tbi'})
+
+        # configure the required resources
+        vep_job.image(image_path('vep_110')).cpu(4).memory('highmem')
+
+        # gcsfuse works only with the root bucket, without prefix:
+        vep_mount_path = to_path(reference_path('vep_110_mount'))
+        data_mount = to_path(f'/{vep_mount_path.drive}')
+        vep_job.cloudfuse(vep_mount_path.drive, str(data_mount), read_only=True)
+        vep_dir = data_mount / '/'.join(vep_mount_path.parts[2:])
+
+        vep_job.command(f'FASTA={vep_dir}/vep/homo_sapiens/*/Homo_sapiens.GRCh38*.fa.gz && echo $FASTA')
+        vep_job.command(
+            f'vep --format vcf --vcf --compress_output bgzip --no_stats --fork 4 --dir_cache {vep_dir}/vep/ '
+            f'-o {vep_job.vcf["vcf.bgz"]} '  # type: ignore
+            f'-i {vcf} --protein --species homo_sapiens --cache --offline --assembly GRCh38 --fa ${{FASTA}}',
+        )
+        vep_job.command(f'tabix -p vcf {vep_job.vcf["vcf.bgz"]}')  # type: ignore
+
+        if isinstance(output_dir, str):
+            get_batch().write_output(vep_job.vcf, os.path.join(output_dir, chrom))
+
+        # keep a list of the in-batch VCF paths
+        ordered_annotated.append(vep_job.vcf["vcf.bgz"])  # type: ignore
+
+    return ordered_annotated
+
+
+def split_and_annotate_vcf(vcf_in: str | ResourceFile, out_vcf: str):
+    """
+    take a path to a VCF (in GCP), or a ResourceFile (with implicit accompanying index)
+    split that VCF into separate chromosomes (dumb split, but fine for small inputs)
+    annotate each chromosomal region separately, then merge the result into a single output
+
+    Args:
+        vcf_in (str): path to an input VCF file, or a pre-localised ResourceFile
+        out_vcf (str): path to final output
+    """
+
+    # allow for situations where this VCF was already a localised Resource
+    if isinstance(vcf_in, str):
+        vcf_in_batch = get_batch().read_input_group(**{'vcf.bgz': vcf_in, 'vcf.bgz.tbi': f'{vcf_in}.tbi'})['vcf.bgz']
+    else:
+        vcf_in_batch = vcf_in
+
+    # I want to store the fragmented VCFs in tmp
+    vcf_fragments_dir = output_path('vcf_fragments', category='tmp')
+
+    # split the whole vcf into chromosomes, but keep accurate ordering for all chunks
+    ordered_output_vcfs: list[ResourceFile] = split_vcf_by_chromosome(
+        localised_vcf=vcf_in_batch,
+        output_dir=vcf_fragments_dir,
+    )
+
+    # new path, also in tmp
+    annotated_tmpdir = output_path('annotated_vcf_fragments', category='tmp')
+
+    ordered_annotated: list[ResourceFile] = annotate_localised_vcfs(ordered_output_vcfs, output_dir=annotated_tmpdir)
+
+    # now merge them all
+    merged_vcf_in_batch = naive_merge_vcfs(ordered_annotated, output_file=out_vcf, vcfs_localised=True)  # type: ignore
```

Comment:
Is bcftools concat a better option here? I guess merge works if there are no samples 🤷‍♂️ .

Reply:
BCFtools concat would be a much better option... I've added a second method to run a concat instead of a merge, and called the concat version in this workflow